sql: forbid TIMETZ during mixed 19.2/20.1 node clusters

otan opened the pull request:
Resolves https://github.com/cockroachdb/cockroach/issues/47110.

Release note (bug fix): This PR fixes a bug preventing clusters from
creating TIMETZ columns before they accept 20.1 without downgrading.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/47156)
<!-- Reviewable:end -->

otan committed:
sql: forbid TIMETZ during mixed 19.2/20.1 node clusters

Release note (bug fix): This PR fixes a bug preventing clusters from
creating TIMETZ columns before they accept 20.1 without downgrading.

rohany commented:
can you verify that spas's bug isn't reproducible with this change

otan commented:
bors r=rohany

i'll give it a go, but i think this should be a backport for sure so expediting it.
